src/main/java/ch/njol/skript/expressions/ExprMOTD.java

```diff
@@ -0,0 +1,127 @@
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.ScriptLoader;
+import ch.njol.util.coll.CollectionUtils;
+import ch.njol.util.Kleenean;
+import com.destroystokyo.paper.event.server.PaperServerListPingEvent;
+import org.bukkit.Bukkit;
+import org.bukkit.event.Event;
+import org.bukkit.event.server.ServerListPingEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("MOTD")
+@Description({"The message of the day in the server list. " +
+		"This can be changed in a <a href='events.html#server_list_ping'>server list ping</a> event only.",
+		"'default MOTD' returns the default MOTD always and can't be changed."})
+@Examples({"on server list ping:",
+		"	set the motd to \"Join now!\""})
+@Since("INSERT VERSION")
+public class ExprMOTD extends SimpleExpression<String> {
+
+	static {
+		Skript.registerExpression(ExprMOTD.class, String.class, ExpressionType.SIMPLE, "[the] [(1¦default)|(2¦shown|displayed)] (MOTD|message of [the] day)");
+	}
+
+	@SuppressWarnings("null")
+	private Kleenean delay;
+
+	private boolean isDefault;
+
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		boolean isServerPingEvent = ScriptLoader.isCurrentEvent(ServerListPingEvent.class) ||
+				(Skript.classExists("com.destroystokyo.paper.event.server.PaperServerListPingEvent") && ScriptLoader.isCurrentEvent(PaperServerListPingEvent.class));
+		if (parseResult.mark == 2 && !isServerPingEvent) {
+			Skript.error("The 'shown' MOTD expression can't be used outside of a server list ping event");
+			return false;
+		}
+		isDefault = (parseResult.mark == 0 && !isServerPingEvent) || parseResult.mark == 1;
+		delay = isDelayed;
+		return true;
+	}
+
+	@Override
+	public boolean isSingle() {
+		return true;
+	}
+
+	@Override
+	@Nullable
+	public String[] get(Event e) {
+		if (isDefault)
+			return CollectionUtils.array(Bukkit.getMotd());
+		else
+			return CollectionUtils.array(((ServerListPingEvent) e).getMotd());
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(Changer.ChangeMode mode) {
+		if (!isDefault) {
+			if (delay == Kleenean.TRUE) {
+				Skript.error("Can't change the MOTD anymore after the server list ping event has already passed");
+				return null;
+			}
+			if (mode == Changer.ChangeMode.SET || mode == Changer.ChangeMode.DELETE || mode == Changer.ChangeMode.RESET)
+				return CollectionUtils.array(String.class);
+		}
+		return null;
+	}
+
+	@Override
+	@SuppressWarnings("null")
+	public void change(Event e, @Nullable Object[] delta, Changer.ChangeMode mode) {
+		switch (mode){
+			case SET:
+				((ServerListPingEvent) e).setMotd((String) delta[0]);
+				break;
+			case DELETE:
+				((ServerListPingEvent) e).setMotd("");
+				break;
+			case RESET:
+				((ServerListPingEvent) e).setMotd(Bukkit.getMotd());
```

Comment:
What is the value of `Bukkit.getMotd()`, is it defined in Bukkit's config?

Also, IMHO it would look nicer if you cast it only in one place, but not really something life changing, only an opinion

Reply:
server.properties lul